jtec-core/src/main/java/edu/tum/sse/jtec/reporting/LCOVReportParser.java

```diff
@@ -0,0 +1,147 @@
+package edu.tum.sse.jtec.reporting;
+
+import com.github.javaparser.Range;
+import com.github.javaparser.StaticJavaParser;
+import com.github.javaparser.ast.CompilationUnit;
+import com.github.javaparser.ast.body.ClassOrInterfaceDeclaration;
+import com.github.javaparser.ast.body.MethodDeclaration;
+import edu.tum.sse.jtec.util.IOUtils;
+import org.apache.commons.io.FileUtils;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.Optional;
+import java.util.regex.MatchResult;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * Parse a {@link TestReport} into an {@link LCOVReport}.
+ */
+public class LCOVReportParser {
+    /**
+     * The root directory of the test report's project
+     */
+    private File baseDir;
+
+    /**
+     * Extensions of project files to consider in LCOV report
+     */
+    private final String[] FILE_EXTENSIONS = {"java"};
+
+    /**
+     * Project files to add to LCOV report
+     */
+    private Collection<File> sourceFiles;
+
+    /**
+     * Regex for parsing entities into their components (package.name.Class#method()returnType)
+     */
+    final static Pattern entityRegex = Pattern.compile("^([^#(]*)\\.([^#(]*)(?:#(\\S*)\\(.+)?$");
+
+    public LCOVReportParser(File baseDir) {
+        setBaseDir(baseDir);
+    }
+
+    public File getBaseDir() {
+        return baseDir;
+    }
+
+    public void setBaseDir(File baseDir) {
+        this.baseDir = baseDir;
+        this.sourceFiles = FileUtils.listFiles(baseDir, FILE_EXTENSIONS, true);
```

Comment:
Instead of this I would go for something similar as here: https://github.com/tum-i4/JTeC/blob/main/jtec-core/src/main/java/edu/tum/sse/jtec/reporting/ReportGenerator.java#L43 
This doesn't require an extra dependency (commons-io) and is almost as nice ;-)